File: ejemplos-consola/aprender-javascript/01-variables-y-tipos.js
// ============================================
// 1. VARIABLES Y TIPOS DE DATOS
// ============================================

console.log("=".repeat(50));
console.log("1. VARIABLES Y TIPOS DE DATOS");
console.log("=".repeat(50));

// Variables con let (puede cambiar)
let edad = 25;
let cantidad = 100;
console.log(`Edad: ${edad}, Tipo: ${typeof edad}`);

// Variables con const (no puede cambiar)
const PI = 3.14159;
const nombre = "Juan";
console.log(`PI: ${PI}, Tipo: ${typeof PI}`);
console.log(`Nombre: ${nombre}, Tipo: ${typeof nombre}`);

// Números
let precio = 19.99;
let temperatura = 36.5;
console.log(`Precio: ${precio}, Tipo: ${typeof precio}`);

// Cadenas de texto (strings)
let apellido = 'Pérez';
let mensaje = `Este es un mensaje
de múltiples líneas`;
console.log(`Mensaje: ${mensaje}`);

// Booleanos
let esActivo = true;
let tienePermiso = false;
console.log(`Es activo: ${esActivo}, Tipo: ${typeof esActivo}`);

// null y undefined
let valor = null;
let sinValor = undefined;
console.log(`Valor: ${valor}, Tipo: ${typeof valor}`);
console.log(`Sin valor: ${sinValor}, Tipo: ${typeof sinValor}`);

// Conversión de tipos
let numeroTexto = "123";
let numeroEntero = parseInt(numeroTexto);
let numeroDecimal = parseFloat(numeroTexto);
let textoNumero = String(123);
console.log(`Conversión: '${numeroTexto}' -> ${numeroEntero}`);

// Template strings (interpolación)
let nombreCompleto = `${nombre} ${apellido}`;
console.log(`Nombre completo: ${nombreCompleto}`);

File: ejemplos-consola/aprender-python/01-variables-y-tipos.py
# ============================================
# 1. VARIABLES Y TIPOS DE DATOS
# ============================================

print("=" * 50)
print("1. VARIABLES Y TIPOS DE DATOS")
print("=" * 50)

# Números enteros
edad = 25
cantidad = 100
print(f"Edad: {edad}, Tipo: {type(edad)}")

# Números decimales (float)
precio = 19.99
temperatura = 36.5
print(f"Precio: {precio}, Tipo: {type(precio)}")

# Cadenas de texto (strings)
nombre = "Juan"
apellido = 'Pérez'
mensaje = """Este es un mensaje
de múltiples líneas"""
print(f"Nombre: {nombre}, Tipo: {type(nombre)}")
print(f"Mensaje: {mensaje}")

# Booleanos
es_activo = True
tiene_permiso = False
print(f"Es activo: {es_activo}, Tipo: {type(es_activo)}")

# None (equivalente a null)
valor = None
print(f"Valor: {valor}, Tipo: {type(valor)}")

# Conversión de tipos
numero_texto = "123"
numero_entero = int(numero_texto)
numero_decimal = float(numero_texto)
texto_numero = str(123)
print(f"Conversión: '{numero_texto}' -> {numero_entero}")

